cohort-engine/src/main/java/com/ibm/cohort/engine/measure/MeasureReportParameterHelper.java

```diff
@@ -0,0 +1,107 @@
+package com.ibm.cohort.engine.measure;
+
+import java.math.BigDecimal;
+import java.time.ZoneId;
+import java.util.TimeZone;
+
+import org.hl7.fhir.instance.model.api.IBaseDatatype;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.DateTimeType;
+import org.hl7.fhir.r4.model.Period;
+import org.opencds.cqf.cql.engine.fhir.converter.FhirTypeConverter;
+import org.opencds.cqf.cql.engine.fhir.converter.FhirTypeConverterFactory;
+import org.opencds.cqf.cql.engine.runtime.Code;
+import org.opencds.cqf.cql.engine.runtime.Concept;
+import org.opencds.cqf.cql.engine.runtime.Date;
+import org.opencds.cqf.cql.engine.runtime.DateTime;
+import org.opencds.cqf.cql.engine.runtime.Interval;
+import org.opencds.cqf.cql.engine.runtime.Quantity;
+import org.opencds.cqf.cql.engine.runtime.Ratio;
+import org.opencds.cqf.cql.engine.runtime.Time;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ca.uhn.fhir.context.FhirVersionEnum;
+import ca.uhn.fhir.model.api.TemporalPrecisionEnum;
+
+
+public class MeasureReportParameterHelper {
```

Comment:
Is this logic possibly more general purpose than just for MeasureReport parameters. It feels like this could be FhirTypeConverter in our namespace or something like that.

Reply:
I agree with it being potentially more general purpose, but I was nervous extending any  of the `FhirTypeConverter` classes in case they had some built-in calls happening behind the scenes that would make it tough control some of the output types (like DateTimes).

I went with wrapping the converter instead. If you would want to sit down and talk about what this might look like as a `FhirTypeConverter`, I'd be game.